service/src/main/java/org/apache/griffin/core/job/JobController.java

```diff
@@ -120,8 +124,13 @@ public JobHealth getHealthInfo() {
     }
 
     @RequestMapping(value = "/jobs/trigger/{id}", method = RequestMethod.POST)
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    public void triggerJob(@PathVariable("id") Long id) throws SchedulerException {
-        jobService.triggerJobById(id);
+    @ResponseStatus(HttpStatus.OK)
+    public Map<String, Object> triggerJob(@PathVariable("id") Long id, @RequestBody(required = false) String request) throws SchedulerException {
+        return Collections.singletonMap("triggerKey", jobService.triggerJobById(id));
+    }
+
+    @RequestMapping(value = "jobs/trigger/{triggerKey:.+}", method = RequestMethod.GET)
```

Comment:
it would be semantically more correct to have it as different "collection", because ids are completely different